### PR TITLE
Add allowed_exceptions parameter to Sphinx.emit() (refs: #7683)

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature

### Purpose
- refs: #7683 
- This allows handlers to raise specified exceptions.

### CODE REVIEW

File: sphinx/application.py [hunk 1]
```diff
@@ -436,20 +436,29 @@ def disconnect(self, listener_id: int) -> None:
...
         """Emit *event* and pass *arguments* to the callback functions.
 
         Return the return values of all callbacks as a list.  Do not emit core
         Sphinx events in extensions!
+
+        .. versionchanged:: 3.1
+
+           Added *allowed_exceptions* to specify path-through exceptions
         """
         return self.events.emit(event, *args)
```
Comment (jakobandersen) [inline sphinx/application.py]: I guess the ``allowed_exceptions`` should be passed on :-).

Comment (tk0miya) [reply sphinx/application.py]: oh, thanks!

**Comment by jakobandersen** [comment]
Except for the comment it looks good to me. I have a modified version of #7683 based on this PR ready afterwards.

**Comment by tk0miya** [comment]
Merged!

**Pull request merged**